ukis_pysat/data.py

```diff
@@ -160,6 +160,63 @@ def query_metadata(self, platform, date, aoi, cloud_cover=None):
         # construct MetadataCollection from list of Metadata objects
         return MetadataCollection([self.construct_metadata(meta_src=m, platform=platform) for m in meta_src])
 
+    def query_metadata_srcid(self, platform, srcid):
+        """Queries satellite image metadata from data source by srcid.
+
+        :param platform: Image platform (<enum 'Platform'>).
+        :param srcid: Srcid of a specific product which is essentially its name (String).
+        :returns: Metadata of product that matches srcid (MetadataCollection object).
+        """
+        if self.src == Datahub.File:
+            # get all json files in datadir that match substr
+            meta_files = sorted(
+                [
+                    Path(dp).joinpath(f)
+                    for dp, dn, filenames in os.walk(self.api)
+                    for substr in self.api_substr
+                    for f in filenames
+                    if f.endswith(".json") and substr in f
+                ],
+                key=lambda path: str(path).lower(),
+            )
+
+            # filter metadata json files by srcid and construct MetadataCollection
+            meta_src = []
+            for meta_file in meta_files:
+                with open(meta_file) as f:
+                    m = json.load(f)
+                    try:
+                        self.construct_metadata(meta_src=m, platform=platform)
+                    except (json.decoder.JSONDecodeError, LookupError, TypeError) as e:
+                        raise ValueError(f"{meta_file.name} not a valid metadata file. {e}.")
+                    m_platform = m["properties"]["platformname"]
+                    m_srcid = m["properties"]["srcid"]
+                    if m_platform == platform.value and m_srcid == srcid:

```

Comment:
Can we not filter this before we did all the other stuff?

Reply:
I can't see how. We first need to load the json files and construct the metadata objects to get the metadata attributes for the filter. Filtering directly on the filename would be possible but not stable enough (e.g. in case the filenames where altered externally and do not contain the srcid anymore).